pkg/authorization/authconfig.go

```diff
@@ -132,92 +133,103 @@ func (k *annotationAuthTypeDetector) Detect(_ context.Context, res *unstructured
 	return spi.Anonymous, nil
 }
 
-type expressionHostExtractor struct {
-	paths []string
-}
-
-func NewExpressionHostExtractor(paths []string) spi.HostExtractor {
-	return &expressionHostExtractor{paths: paths}
-}
+func UnifiedHostExtractor(extractors ...spi.HostExtractor) spi.HostExtractor { //nolint:gocognit //reason Inlined functions to avoid surface pollution
+	unique := func(in []string) []string {
+		store := map[string]bool{}
 
-func (k *expressionHostExtractor) Extract(target *unstructured.Unstructured) ([]string, error) {
-	hosts := []string{}
-
-	for _, path := range k.paths {
-		splitPath := strings.Split(path, ".")
-		extractedHosts, err := extractHosts(target, splitPath)
-
-		if err != nil {
-			return nil, fmt.Errorf("failed to extract hosts at path %s: %w", path, err)
+		for _, v := range in {
+			store[v] = true
 		}
 
-		parsedHosts, errAppend := appendHosts(hosts, extractedHosts...)
+		keys := make([]string, len(store))
 
-		if errAppend != nil {
-			return nil, fmt.Errorf("failed to append hosts %v: %w", extractedHosts, errAppend)
+		i := 0
+
+		for v := range store {
+			keys[i] = v
+			i++
 		}
 
-		hosts = parsedHosts
+		return keys
 	}
 
-	u := unique(hosts)
-	if len(u) == 0 {
-		return []string{"unknown.host.com"}, nil
+	isURI := func(host string) bool {
+		return strings.HasPrefix(host, "http://") || strings.HasPrefix(host, "https://")
```

Comment:
Little nit, but maybe we should be more accurate calling it `isHTTP` instead?

Reply:
Well, the intent is to determine if we can call url.Parse which require a protocol. So in that sense a better name would have been isURL(in hindsight..). We currently only "know of" http/http and it's a bit of a short cut to a proper check you can say.

Reply:
Fair enough, how about `isSupportedURL` then?

Reply:
made it isURL. The intent of the whole thing is to extract Hosts. If it's a URL we need to parse it to find the host. SupportedURL sounds more like we're looking for a type of URL which is not the case, even tho the impl is a bit too specific to http atm. The function should be named after it's intent not the current impl. 

https://github.com/maistra/odh-platform/pull/49/commits/f7fc2725c6fc435b15ce06106a2864e0e5ee4b52